falcon/request_helpers.py

```diff
@@ -129,3 +134,271 @@ def readlines(self, hint=None):
         """
 
         return self._read(hint, self.stream.readlines)
+
+
+class ParamProxy(object):
+    """
+    Provide `get_*` methods for both query-string parameters and
+    request body parameters.
+    """
+
+    def __init__(self, request, source):
+        self.req = request
+        self.source = source
+
+    def get(self, name, required=False, store=None):
+        """Return the raw value of a query string parameter as a string.
+
+        Note:
+            Similar to the way multiple keys in form data is handled,
+            if a query parameter is assigned a comma-separated list of
+            values (e.g., 'foo=a,b,c'), only one of those values will be
+            returned, and it is undefined which one. Use
+            `req.get_param_as_list()` to retrieve all the values.
+
+        Args:
+            name (str): Parameter name, case-sensitive (e.g., 'sort').
+            required (bool, optional): Set to ``True`` to raise
+                ``HTTPBadRequest`` instead of returning ``None`` when the
+                parameter is not found (default ``False``).
+            store (dict, optional): A ``dict``-like object in which to place
+                the value of the param, but only if the param is present.
+
+        Returns:
+            str: The value of the param as a string, or ``None`` if param is
+                not found and is not required.
+
+        Raises:
+            HTTPBadRequest: A required param is missing from the request.
+
+        """
+
+        params = self.source
+
+        # PERF: Use if..in since it is a good all-around performer; we don't
+        #       know how likely params are to be specified by clients.
+        if name in params:
+            # NOTE(warsaw): If the key appeared multiple times, it will be
+            # stored internally as a list.  We do not define which one
+            # actually gets returned, but let's pick the last one for grins.
+            param = params[name]
+            if isinstance(param, list):
+                param = param[-1]
+
+            if store is not None:
+                store[name] = param
+
+            return param
+
+        if not required:
+            return None
+
+        raise HTTPMissingParam(name)
+
+    def get_as_int(self, name,
```

Comment:
I was about to suggest to drop the `get`, and have `req.params.as_int('amount')` but your suggestion is more consistent with the simple `get` method, so I'm +1.